### ISSUE TITLE
`collections.OrderdDict` causes file analysis to hang

### ISSUE THREAD

**Issue by fynnsu** [microsoft/pyright#5594]
## Environment data

-   Language Server version: 2023.7.40 (pyright b917b7a5)
-   OS and version: Ubuntu 22.04.1
-   Python version (& distribution if applicable, e.g. Anaconda): 3.9.17 (Anaconda)

Note: I created a fresh anaconda env for this test (using `conda create --name empty python=3.9`) but this issue also occurred in other python environments I tried (including a non-anaconda `python==3.10` env)

## Code SnippetA

```python
import collections

collections.OrderedDict([
    ("key1", collections.OrderedDict.__getitem__),
    ("key2", 3),
])
```

## Repro Steps

1. Create a python file with the above snippet in a vscode workspace
2. Run `Developer: Reload Window`

## Expected behavior

File Analysis should finish

## Actual behavior

File Analysis hangs indefinitely. Note in the logs below `IDX(2)` finishes but `BG(1)` doesn't

## Logs

```
2023-07-28 10:38:30.087 [info] [Info  - 10:38:30 AM] (1531026) Pylance language server 2023.7.40 (pyright b917b7a5) starting
2023-07-28 10:38:30.088 [info] [Info  - 10:38:30 AM] (1531026) Server root directory: /home/fynnsu/.vscode-server/extensions/ms-python.vscode-pylance-2023.7.40/dist
2023-07-28 10:38:30.088 [info] [Info  - 10:38:30 AM] (1531026) Starting service instance "empty"
2023-07-28 10:38:30.162 [info] (1531026) No configuration file found.
2023-07-28 10:38:30.162 [info] (1531026) No pyproject.toml file found.
2023-07-28 10:38:30.162 [info] [Info  - 10:38:30 AM] (1531026) Setting pythonPath for service "empty": "/home/fynnsu/miniconda3/envs/empty/bin/python"
2023-07-28 10:38:30.205 [info] [Info  - 10:38:30 AM] (1531026) Assuming Python version 3.9
2023-07-28 10:38:30.206 [info] (1531026) Assuming Python platform Linux
2023-07-28 10:38:30.312 [info] [Info  - 10:38:30 AM] (1531026) Search paths for /home/fynnsu/projects/tmp/empty
2023-07-28 10:38:30.313 [info] [Info  - 10:38:30 AM] (1531026)   /home/fynnsu/.vscode-server/extensions/ms-python.vscode-pylance-2023.7.40/dist/typeshed-fallback/stdlib
2023-07-28 10:38:30.313 [info] [Info  - 10:38:30 AM] (1531026)   /home/fynnsu/projects/tmp/empty
2023-07-28 10:38:30.313 [info] [Info  - 10:38:30 AM] (1531026)   /home/fynnsu/projects/tmp/empty/typings
2023-07-28 10:38:30.313 [info] [Info  - 10:38:30 AM] (1531026)   /home/fynnsu/.vscode-server/extensions/ms-python.vscode-pylance-2023.7.40/dist/typeshed-fallback/stubs/...
2023-07-28 10:38:30.314 [info] [Info  - 10:38:30 AM] (1531026)   /home/fynnsu/.vscode-server/extensions/ms-python.vscode-pylance-2023.7.40/dist/bundled/stubs
2023-07-28 10:38:30.314 [info] [Info  - 10:38:30 AM] (1531026)   /home/fynnsu/miniconda3/envs/empty/lib/python3.9
2023-07-28 10:38:30.314 [info] [Info  - 10:38:30 AM] (1531026)   /home/fynnsu/miniconda3/envs/empty/lib/python3.9/lib-dynload
2023-07-28 10:38:30.315 [info] [Info  - 10:38:30 AM] (1531026)   /home/fynnsu/miniconda3/envs/empty/lib/python3.9/site-packages
2023-07-28 10:38:30.315 [info] [Info  - 10:38:30 AM] (1531026) Adding fs watcher for library directories:
 /home/fynnsu/miniconda3/envs/empty/lib/python3.9
/home/fynnsu/miniconda3/envs/empty/lib/python3.9/lib-dynload
/home/fynnsu/miniconda3/envs/empty/lib/python3.9/site-packages
2023-07-28 10:38:30.316 [info] [Info  - 10:38:30 AM] (1531026) Adding fs watcher for directories:
 /home/fynnsu/projects/tmp/empty
2023-07-28 10:38:30.317 [info] (1531026) Searching for source files
2023-07-28 10:38:30.322 [info] [Info  - 10:38:30 AM] (1531026) Found 1 source file
2023-07-28 10:38:30.458 [info] (1531026) [FG] parsing: /home/fynnsu/projects/tmp/empty/test.py (87ms)
2023-07-28 10:38:30.721 [info] (1531026) [FG] parsing: /home/fynnsu/.vscode-server/extensions/ms-python.vscode-pylance-2023.7.40/dist/typeshed-fallback/stdlib/builtins.pyi [fs read 3ms] (261ms)
2023-07-28 10:38:30.802 [info] (1531026) [FG] binding: /home/fynnsu/.vscode-server/extensions/ms-python.vscode-pylance-2023.7.40/dist/typeshed-fallback/stdlib/builtins.pyi (78ms)
2023-07-28 10:38:30.802 [info] (1531026) [FG] binding: /home/fynnsu/projects/tmp/empty/test.py (0ms)
2023-07-28 10:38:30.855 [info] [Info  - 10:38:30 AM] (1531026) Background analysis(1) root directory: /home/fynnsu/.vscode-server/extensions/ms-python.vscode-pylance-2023.7.40/dist
2023-07-28 10:38:30.858 [info] [Info  - 10:38:30 AM] (1531026) Background analysis(1) started
2023-07-28 10:38:30.864 [info] (1531026) Background analysis message: setConfigOptions
2023-07-28 10:38:30.865 [info] (1531026) Background analysis message: setImportResolver
2023-07-28 10:38:30.866 [info] (1531026) Background analysis message: ensurePartialStubPackages
2023-07-28 10:38:30.965 [info] (1531026) Background analysis message: setConfigOptions
2023-07-28 10:38:30.971 [info] (1531026) Background analysis message: setTrackedFiles
2023-07-28 10:38:30.990 [info] (1531026) Background analysis message: markAllFilesDirty
2023-07-28 10:38:30.991 [info] (1531026) Background analysis message: setFileOpened
2023-07-28 10:38:30.991 [info] (1531026) Background analysis message: analyze
2023-07-28 10:38:31.049 [info] (1531026) [BG(1)] analyzing: /home/fynnsu/projects/tmp/empty/test.py ...
2023-07-28 10:38:31.049 [info] (1531026) [BG(1)]   parsing: /home/fynnsu/projects/tmp/empty/test.py (65ms)
2023-07-28 10:38:31.076 [info] [Info  - 10:38:31 AM] (1531026) Indexer background runner(2) root directory: /home/fynnsu/.vscode-server/extensions/ms-python.vscode-pylance-2023.7.40/dist (index)
2023-07-28 10:38:31.076 [info] [Info  - 10:38:31 AM] (1531026) Indexing(2) started
2023-07-28 10:38:31.195 [info] (1531026) [IDX(2)] scan packages /home/fynnsu/projects/tmp/empty ...
2023-07-28 10:38:31.195 [info] (1531026) [IDX(2)]   read stdlib indices (108ms)
2023-07-28 10:38:31.240 [info] (1531026) [BG(1)]   parsing: /home/fynnsu/.vscode-server/extensions/ms-python.vscode-pylance-2023.7.40/dist/typeshed-fallback/stdlib/builtins.pyi [fs read 3ms] (191ms)
2023-07-28 10:38:31.306 [info] (1531026) [BG(1)]   binding: /home/fynnsu/.vscode-server/extensions/ms-python.vscode-pylance-2023.7.40/dist/typeshed-fallback/stdlib/builtins.pyi (63ms)
2023-07-28 10:38:31.306 [info] (1531026) [BG(1)]   binding: /home/fynnsu/projects/tmp/empty/test.py (0ms)
2023-07-28 10:38:31.375 [info] (1531026) [BG(1)]   checking: /home/fynnsu/projects/tmp/empty/test.py ...
2023-07-28 10:38:31.375 [info] (1531026) [BG(1)]     parsing: /home/fynnsu/.vscode-server/extensions/ms-python.vscode-pylance-2023.7.40/dist/typeshed-fallback/stdlib/typing.pyi [fs read 2ms] (54ms)
2023-07-28 10:38:31.394 [info] (1531026) [BG(1)]     binding: /home/fynnsu/.vscode-server/extensions/ms-python.vscode-pylance-2023.7.40/dist/typeshed-fallback/stdlib/typing.pyi (18ms)
2023-07-28 10:38:31.409 [info] (1531026) [BG(1)]     parsing: /home/fynnsu/.vscode-server/extensions/ms-python.vscode-pylance-2023.7.40/dist/typeshed-fallback/stdlib/typing_extensions.pyi [fs read 1ms] (12ms)
2023-07-28 10:38:31.416 [info] (1531026) [BG(1)]     binding: /home/fynnsu/.vscode-server/extensions/ms-python.vscode-pylance-2023.7.40/dist/typeshed-fallback/stdlib/typing_extensions.pyi (7ms)
2023-07-28 10:38:31.440 [info] (1531026) [BG(1)]     parsing: /home/fynnsu/.vscode-server/extensions/ms-python.vscode-pylance-2023.7.40/dist/typeshed-fallback/stdlib/_typeshed/__init__.pyi [fs read 0ms] (5ms)
2023-07-28 10:38:31.448 [info] (1531026) [BG(1)]     binding: /home/fynnsu/.vscode-server/extensions/ms-python.vscode-pylance-2023.7.40/dist/typeshed-fallback/stdlib/_typeshed/__init__.pyi (7ms)
2023-07-28 10:38:31.459 [info] (1531026) [BG(1)]     parsing: /home/fynnsu/.vscode-server/extensions/ms-python.vscode-pylance-2023.7.40/dist/typeshed-fallback/stdlib/abc.pyi [fs read 0ms] (2ms)
2023-07-28 10:38:31.460 [info] (1531026) [BG(1)]     binding: /home/fynnsu/.vscode-server/extensions/ms-python.vscode-pylance-2023.7.40/dist/typeshed-fallback/stdlib/abc.pyi (1ms)
2023-07-28 10:38:31.501 [info] (1531026) [BG(1)]     parsing: /home/fynnsu/.vscode-server/extensions/ms-python.vscode-pylance-2023.7.40/dist/typeshed-fallback/stdlib/collections/__init__.pyi [fs read 0ms] (38ms)
2023-07-28 10:38:31.505 [info] (1531026) [BG(1)]     binding: /home/fynnsu/.vscode-server/extensions/ms-python.vscode-pylance-2023.7.40/dist/typeshed-fallback/stdlib/collections/__init__.pyi ...
2023-07-28 10:38:31.506 [info] (1531026) [BG(1)]       parsing: /home/fynnsu/.vscode-server/extensions/ms-python.vscode-pylance-2023.7.40/dist/typeshed-fallback/stdlib/_collections_abc.pyi [fs read 0ms] (4ms)
2023-07-28 10:38:31.507 [info] (1531026) [BG(1)]       binding: /home/fynnsu/.vscode-server/extensions/ms-python.vscode-pylance-2023.7.40/dist/typeshed-fallback/stdlib/_collections_abc.pyi (1ms)
2023-07-28 10:38:31.521 [info] (1531026) [BG(1)]     binding: /home/fynnsu/.vscode-server/extensions/ms-python.vscode-pylance-2023.7.40/dist/typeshed-fallback/stdlib/collections/__init__.pyi (20ms)
2023-07-28 10:38:31.540 [info] (1531026) [BG(1)]     parsing: /home/fynnsu/.vscode-server/extensions/ms-python.vscode-pylance-2023.7.40/dist/typeshed-fallback/stdlib/collections/abc.pyi [fs read 0ms] (3ms)
2023-07-28 10:38:31.541 [info] (1531026) [BG(1)]     binding: /home/fynnsu/.vscode-server/extensions/ms-python.vscode-pylance-2023.7.40/dist/typeshed-fallback/stdlib/collections/abc.pyi (0ms)
2023-07-28 10:38:31.606 [info] (1531026) [IDX(2)] scan packages /home/fynnsu/projects/tmp/empty (512ms)
2023-07-28 10:38:31.606 [info] [Info  - 10:38:31 AM] (1531026) scanned(2) 15 files over 1 exec env
2023-07-28 10:38:31.694 [info] (1531026) [IDX(2)] index packages /home/fynnsu/projects/tmp/empty ...
2023-07-28 10:38:31.694 [info] (1531026) [IDX(2)]   index execution environment /home/fynnsu/projects/tmp/empty ...
2023-07-28 10:38:31.694 [info] (1531026) [IDX(2)]     indexing: /home/fynnsu/miniconda3/envs/empty/lib/python3.9/concurrent/__init__.py [skipped: no '__all__' defined] (0ms)
2023-07-28 10:38:31.694 [info] (1531026) [IDX(2)]     indexing: /home/fynnsu/miniconda3/envs/empty/lib/python3.9/distutils/__init__.py [skipped: no '__all__' defined] (1ms)
2023-07-28 10:38:31.695 [info] (1531026) [IDX(2)]     indexing: /home/fynnsu/miniconda3/envs/empty/lib/python3.9/idlelib/__init__.py [skipped: no '__all__' defined] (0ms)
2023-07-28 10:38:31.695 [info] (1531026) [IDX(2)]     indexing: /home/fynnsu/miniconda3/envs/empty/lib/python3.9/lib2to3/__init__.py [skipped: no '__all__' defined] (0ms)
2023-07-28 10:38:31.695 [info] (1531026) [IDX(2)]     indexing: /home/fynnsu/miniconda3/envs/empty/lib/python3.9/pydoc_data/__init__.py [skipped: no '__all__' defined] (0ms)
2023-07-28 10:38:31.696 [info] (1531026) [IDX(2)]     indexing: /home/fynnsu/miniconda3/envs/empty/lib/python3.9/test/__init__.py [skipped: no '__all__' defined] (0ms)
2023-07-28 10:38:31.696 [info] (1531026) [IDX(2)]     indexing: /home/fynnsu/miniconda3/envs/empty/lib/python3.9/turtledemo/__init__.py [skipped: no '__all__' defined] (0ms)
2023-07-28 10:38:31.696 [info] (1531026) [IDX(2)]     indexing: /home/fynnsu/miniconda3/envs/empty/lib/python3.9/urllib/__init__.py [skipped: no '__all__' defined] (1ms)
2023-07-28 10:38:31.696 [info] (1531026) [IDX(2)]     indexing: /home/fynnsu/miniconda3/envs/empty/lib/python3.9/wsgiref/__init__.py [skipped: no '__all__' defined] (0ms)
2023-07-28 10:38:31.771 [info] (1531026) [IDX(2)]     indexing: /home/fynnsu/miniconda3/envs/empty/lib/python3.9/xml/__init__.py ...
2023-07-28 10:38:31.771 [info] (1531026) [IDX(2)]       parsing: /home/fynnsu/miniconda3/envs/empty/lib/python3.9/xml/__init__.py [fs read 0ms] (71ms)
2023-07-28 10:38:31.889 [info] (1531026) [IDX(2)]       parsing: /home/fynnsu/.vscode-server/extensions/ms-python.vscode-pylance-2023.7.40/dist/typeshed-fallback/stdlib/builtins.pyi [fs read 2ms] (115ms)
2023-07-28 10:38:31.911 [info] (1531026) [IDX(2)]       binding: /home/fynnsu/.vscode-server/extensions/ms-python.vscode-pylance-2023.7.40/dist/typeshed-fallback/stdlib/builtins.pyi (21ms)
2023-07-28 10:38:31.911 [info] (1531026) [IDX(2)]       binding: /home/fynnsu/miniconda3/envs/empty/lib/python3.9/xml/__init__.py (0ms)
2023-07-28 10:38:31.912 [info] (1531026) [IDX(2)]     indexing: /home/fynnsu/miniconda3/envs/empty/lib/python3.9/xml/__init__.py [found 0] (216ms)
2023-07-28 10:38:31.912 [info] (1531026) [IDX(2)]     indexing: /home/fynnsu/miniconda3/envs/empty/lib/python3.9/xmlrpc/__init__.py [skipped: no '__all__' defined] (0ms)
2023-07-28 10:38:31.919 [info] (1531026) [IDX(2)]     indexing: /home/fynnsu/miniconda3/envs/empty/lib/python3.9/site-packages/pip/__init__.py ...
2023-07-28 10:38:31.919 [info] (1531026) [IDX(2)]       parsing: /home/fynnsu/miniconda3/envs/empty/lib/python3.9/site-packages/pip/__init__.py [fs read 2ms] (3ms)
2023-07-28 10:38:31.920 [info] (1531026) [IDX(2)]       binding: /home/fynnsu/miniconda3/envs/empty/lib/python3.9/site-packages/pip/__init__.py (0ms)
2023-07-28 10:38:31.920 [info] (1531026) [IDX(2)]     indexing: /home/fynnsu/miniconda3/envs/empty/lib/python3.9/site-packages/pip/__init__.py [found 2] (8ms)
2023-07-28 10:38:31.921 [info] (1531026) [IDX(2)]     indexing: /home/fynnsu/miniconda3/envs/empty/lib/python3.9/site-packages/wheel/__init__.py [skipped: no '__all__' defined] (0ms)
2023-07-28 10:38:31.921 [info] (1531026) [IDX(2)]   index execution environment /home/fynnsu/projects/tmp/empty [found 98 in 4 files] (234ms)
2023-07-28 10:38:31.921 [info] (1531026) [IDX(2)] index packages /home/fynnsu/projects/tmp/empty [found 98 in 1 exec envs] (237ms)
2023-07-28 10:38:31.921 [info] [Info  - 10:38:31 AM] (1531026) indexed(2) 4 files over 1 exec env
2023-07-28 10:38:31.924 [info] [Info  - 10:38:31 AM] (1531026) Indexing finished(2).
```

**Comment by erictraut** [comment]
Thanks for the bug report. This will be addressed in the next release.

**Comment by erictraut** [comment]
This is addressed in pyright 1.1.320, which I just published. It will also be included in a future version of pylance.
